rados/ioctx.go

```diff
@@ -249,19 +251,23 @@ func (ioctx *IOContext) GetPoolStats() (stat PoolStat, err error) {
 
 // GetPoolName returns the name of the pool associated with the I/O context.
 func (ioctx *IOContext) GetPoolName() (name string, err error) {
-	buf := make([]byte, 128)
-	for {
-		ret := C.rados_ioctx_get_pool_name(ioctx.ioctx,
-			(*C.char)(unsafe.Pointer(&buf[0])), C.unsigned(len(buf)))
-		if ret == -C.ERANGE {
-			buf = make([]byte, len(buf)*2)
-			continue
-		} else if ret < 0 {
-			return "", getRadosError(int(ret))
-		}
-		name = C.GoStringN((*C.char)(unsafe.Pointer(&buf[0])), ret)
-		return name, nil
+	var (
+		buf []byte
+		ret C.int
+	)
+	for z := retry.NewSizerEV(128, 8192, errRange); z.Continue(); {
+		buf = make([]byte, z.Size())
+		ret = C.rados_ioctx_get_pool_name(
+			ioctx.ioctx,
+			(*C.char)(unsafe.Pointer(&buf[0])),
+			C.unsigned(len(buf)))
+		err = z.Update(retNegToError(ret))
```

Comment:
It really is, isn't it. :-)
I have been thinking about how to improve error handling and prevent code duplication around errors and had not worked on improving this situation because I thought I'd have a solution to that issue by now. Since I have not, how aboug `getErrorIfNegative`? It retains the 'getError' pattern we already have and adds a clarifying suffix.

Reply:
Renamed.